auto olm test case OCP-23440: the end user can use operator resources

jianzhangbjz opened the pull request:
@bparees @ecordell @njhale As title, please have a review. One question, what's the best solution to switch between `cluster-admin user` and `common user`? Do we have any code implement it? Thanks! cc: @cuipinghuo @bandrade @scolange @chengzhang1016 @zihantang-rh @emmajiafan

bparees commented:
>  what's the best solution to switch between cluster-admin user and common user

using "oc.AsAdmin()" to perform actions as kubeadmin and just oc.X() to do it as the normal e2e user.

jianzhangbjz commented:
@bparees Got it, thanks! @njhale Could you help review this PR? Thanks!

jianzhangbjz commented:
/assign @bparees @njhale

jianzhangbjz commented:
/test e2e-aws

jianzhangbjz commented:
Try to fix below errors:
```
fail [github.com/openshift/origin/test/extended/operators/olm.go:96]: Unexpected error:
    <*util.ExitError | 0xc004214450>: {
        Cmd: "oc --namespace=e2e-test-olm-23440-fnwcv --config=/tmp/admin.kubeconfig create -f /tmp/e2e-test-olm-23440-fnwcv-config.json",
        StdErr: "error: the namespace from the provided object \"openshift-marketplace\" does not match the namespace \"e2e-test-olm-23440-fnwcv\". You must pass '--namespace=openshift-marketplace' to perform this operation.",
```

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
Add `spec.source` field to cover the latest `csc` changes.
```
fail [github.com/openshift/origin/test/extended/operators/olm.go:96]: Unexpected error:
    <*util.ExitError | 0xc000023dd0>: {
        Cmd: "oc --config=/tmp/admin.kubeconfig create -f /tmp/e2e-test-olm-23440-2ndqf-config.json",
        StdErr: "The CatalogSourceConfig \"test-operator\" is invalid: []: Invalid value: map[string]interface {}{\"apiVersion\":\"operators.coreos.com/v1\", \"kind\":\"CatalogSourceConfig\", \"metadata\":map[string]interface {}{\"creationTimestamp\":\"2019-07-18T03:07:37Z\", \"generation\":1, \"name\":\"test-operator\", \"namespace\":\"openshift-marketplace\", \"uid\":\"3213d429-a909-11e9-9d9d-12dda7ad6924\"}, \"spec\":map[string]interface {}{\"packages\":\"etcd\", \"targetNamespace\":\"e2e-test-olm-23440-2ndqf\"}}: validation failure list:\nspec.source in body is required"
```

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/test e2e-aws

jianzhangbjz commented:
@bparees @njhale Could help review this PR and give approve if it looks good? Thanks!

jianzhangbjz commented:
/test e2e-aws-serial

jianzhangbjz commented:
@bparees @njhale Could help review this PR and give approve if it looks good? Thanks!

jianzhangbjz commented:
@bparees @njhale Could help review this PR and give approve if it looks good? Thanks! It's in `Pending` status for a long time.

bparees commented:
@jianzhangbjz i'm not actually on the OLM team, so should be reviewed by @njhale or @ecordell or someone else from that team.

jianzhangbjz commented:
@bparees Thanks! I thought @njhale @ecordell may be haven't the merging permission.  So, I message you. @njhale @ecordell Could you help have a review add the `/lgtm` label if it looks good? Thanks!

bparees commented:
/approve

now they can definitely merge it :)

ecordell commented on test/extended/operators/olm.go:
```diff
@@ -71,3 +71,57 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+		}
+		time.Sleep(90 * time.Second)
```
Can we poll or watch instead of sleeping?

jianzhangbjz replied:
Yeah, I also don't like to use `sleep`, but I couldn't find a better solution. If we use the `poll`, sometimes, the subscription failed to create, and it will hang there, and I'm not sure the exact rounds we should query.

jianzhangbjz replied:
@ecordell Do you have any suggestions for this step? Thanks!

jianzhangbjz replied:
@ecordell Done, updated it, thanks!

ecordell commented on test/extended/operators/olm.go:
```diff
@@ -71,3 +71,57 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("create").Args("-f", configFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(90 * time.Second)
```
Same - can we poll or watch the etcdcluster's status?

jianzhangbjz replied:
@ecordell Thanks! Replace it with `wait.Poll`.

njhale commented on test/extended/operators/olm.go:
```diff
@@ -71,3 +71,57 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+	var (
+		buildPruningBaseDir = exutil.FixturePath("testdata", "olm")
+		operatorGroup       = filepath.Join(buildPruningBaseDir, "operatorgroup.yaml")
+		catalogSourceConfig = filepath.Join(buildPruningBaseDir, "csc.yaml")
```
Should we be testing CSCs? The type is not provided by OLM and is not required to test if we can install an operator from an existing CatalogSource.

jianzhangbjz replied:
Yes, you're right. We don't need to create CSC for the test in 4.2, but it's a necessary for 4.1. Remove it, thanks!

jianzhangbjz commented on test/extended/operators/olm.go:
```diff
@@ -71,3 +71,57 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 })
+
+// This context will cover test case: OCP-23440, author: jiazha@redhat.com
+var _ = g.Describe("[Feature:Platform] the end user use OLM", func() {
```
Done, thanks!

jianzhangbjz commented on test/extended/operators/olm.go:
```diff
@@ -71,3 +71,57 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+	)
+
+	files := []string{operatorGroup, catalogSourceConfig, etcdSub}
+	g.It(" subscribe the single etcd operator", func() {
```
Done, thanks!

jianzhangbjz commented:
/test e2e-aws

jianzhangbjz commented:
/test e2e-aws-upgrade
/test e2e-aws

jianzhangbjz commented:
@ecordell @njhale Could you help have a review add the /lgtm label if it looks good? Thanks!

jianzhangbjz commented:
@ecordell @njhale Could you help have a review? Thanks!

njhale commented:
Looking good. I think you should probably squash a few of the commits so the history looks a little cleaner.

jianzhangbjz commented:
@njhale Yes, I squashed the history. Thanks! Could you help add the /lgtm label to merge it? Thanks!

njhale commented on test/extended/operators/olm.go:
```diff
@@ -71,3 +74,73 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+			"size":           "3",
+		}
+
+		for k, v := range checkMetrics {
```
You don't really need this loop since `wait.Poll` will churn until timeout, error, or a true return value. You can just check the `phase`, `currentVersion`, and `size` in the closure argument.

jianzhangbjz replied:
@njhale Yes, I know. The destination of this loop just to check that three metrics("phase", "currentVersion", "size") and their expected results("Running", "3.2.13", "3"). What do you suggest?

jianzhangbjz replied:
@njhale I have removed this loop, thanks!

jianzhangbjz commented:
/retest

jianzhangbjz commented:
@bparees @njhale @ecordell Could you help give approval? Thanks!

jianzhangbjz commented:
@njhale @ecordell Could you help add the `lgtm` label? Thanks!

njhale commented:
Thanks for your patience.

/lgtm

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jottofar commented:
@jianzhangbjz unit.log shows "panic: test timed out after 3m0s\n". I got this same error in 3.11PR https://github.com/openshift/origin/pull/23948 and had to both increase test_timeout and run tests serially "go test -p 1". The latter may be all that was really needed.

jianzhangbjz commented:
@jottofar Thank you! I increased the waiting time and added the `go test -p 1`.

jianzhangbjz commented:
/retest

jottofar commented:
build-log.txt:
[WARNING] go test: p flag may be set only once

Does TESTFLAGS also have '-p' set perhaps?

jianzhangbjz commented:
@jottofar Sorry for the late reply. I'm not sure. @bparees Do you have any ideas? Thanks!

bparees commented on hack/test-go.sh:
```diff
@@ -131,7 +131,7 @@ if [[ -n "${junit_report}" ]]; then
     report_file="$( mktemp "${ARTIFACT_DIR}/unit_report_XXXXX" ).xml"
 
-    go test -json ${gotest_flags} ${test_packages} 2>"${test_error_file}" | tee "${JUNIT_REPORT_OUTPUT}" | gotest2junit > "${report_file}"
+    go test -p 1 -json ${gotest_flags} ${test_packages} 2>"${test_error_file}" | tee "${JUNIT_REPORT_OUTPUT}" | gotest2junit > "${report_file}"
```
presumably -p is already being set in gotest_flags so you can't set it again.

jianzhangbjz replied:
@bparees Thanks!

jianzhangbjz commented:
It does PASS running on my laptop, as follows. `/retest`
```console
mac:openshift-tests jianzhang$ oc get clusterversion
NAME      VERSION                             AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.3.0-0.nightly-2019-11-29-051144   True        False         3h32m   Cluster version is 4.3.0-0.nightly-2019-11-29-051144
mac:openshift-tests jianzhang$ ./openshift-tests run all --dry-run | grep "\[Feature:Platform\] an end user use OLM"  | ./openshift-tests run --junit-dir=./ -f -
Dec  2 13:15:59.719: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 13:15:59.721: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
started: (0/1/1) "[Feature:Platform] an end user use OLM can subscribe to the etcd operator [Suite:openshift/conformance/parallel]"

W1202 13:18:03.483704   18845 reflector.go:299] github.com/openshift/origin/pkg/monitor/operator.go:126: watch of *v1.ClusterOperator ended with: too old resource version: 70810 (70863)
W1202 13:18:08.766502   18845 reflector.go:299] github.com/openshift/origin/pkg/monitor/operator.go:126: watch of *v1.ClusterOperator ended with: too old resource version: 70889 (70898)
Dec  2 13:16:09.055: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 13:16:09.057: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 13:16:15.969: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 13:16:15.970: INFO: Waiting up to 30m0s for all (but 100) nodes to be schedulable
Dec  2 13:16:16.792: INFO: Waiting up to 10m0s for all pods (need at least 0) in namespace 'kube-system' to be running and ready
Dec  2 13:16:17.667: INFO: 0 / 0 pods in namespace 'kube-system' are running and ready (0 seconds elapsed)
Dec  2 13:16:17.667: INFO: expected 0 pod replicas in namespace 'kube-system', 0 are Running and Ready.
Dec  2 13:16:17.667: INFO: Waiting up to 5m0s for all daemonsets in namespace 'kube-system' to start
Dec  2 13:16:17.925: INFO: e2e test version: v0.0.0-master+$Format:%h$
Dec  2 13:16:18.182: INFO: kube-apiserver version: v1.16.2
Dec  2 13:16:18.182: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 13:16:18.483: INFO: Cluster IP family: ipv4
[BeforeEach] [Top Level]
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/util/test.go:60
[BeforeEach] [Feature:Platform] an end user use OLM
  /Users/jianzhang/goproject/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:151
STEP: Creating a kubernetes client
Dec  2 13:16:18.492: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
[BeforeEach] [Feature:Platform] an end user use OLM
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/util/client.go:109
Dec  2 13:16:21.045: INFO: configPath is now "/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930"
Dec  2 13:16:21.045: INFO: The user is now "e2e-test-olm-23440-fjdbm-user"
Dec  2 13:16:21.045: INFO: Creating project "e2e-test-olm-23440-fjdbm"
Dec  2 13:16:21.657: INFO: Waiting on permissions in project "e2e-test-olm-23440-fjdbm" ...
Dec  2 13:16:21.908: INFO: Waiting for ServiceAccount "default" to be provisioned...
Dec  2 13:16:22.271: INFO: Waiting for ServiceAccount "deployer" to be provisioned...
Dec  2 13:16:22.681: INFO: Waiting for ServiceAccount "builder" to be provisioned...
Dec  2 13:16:23.089: INFO: Waiting for RoleBinding "system:image-pullers" to be provisioned...
Dec  2 13:16:23.705: INFO: Waiting for RoleBinding "system:image-builders" to be provisioned...
Dec  2 13:16:24.206: INFO: Waiting for RoleBinding "system:deployers" to be provisioned...
Dec  2 13:16:24.831: INFO: Project "e2e-test-olm-23440-fjdbm" has been fully provisioned.
[It] can subscribe to the etcd operator [Suite:openshift/conformance/parallel]
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/operators/olm.go:109
STEP: Cluster-admin user subscribe the operator resource
Dec  2 13:16:24.831: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/Users/jianzhang/43-kubeconfig process --ignore-unknown-parameters=true -f /var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/fixture-testdata-dir223261975/test/extended/testdata/olm/operatorgroup.yaml -p NAME=test-operator NAMESPACE=e2e-test-olm-23440-fjdbm SOURCENAME=community-operators SOURCENAMESPACE=openshift-marketplace'
Dec  2 13:16:26.108: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig create -f /tmp/e2e-test-olm-23440-fjdbm-config.json'
operatorgroup.operators.coreos.com/test-operator created
Dec  2 13:16:27.401: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/Users/jianzhang/43-kubeconfig process --ignore-unknown-parameters=true -f /var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/fixture-testdata-dir223261975/test/extended/testdata/olm/etcd-subscription.yaml -p NAME=test-operator NAMESPACE=e2e-test-olm-23440-fjdbm SOURCENAME=community-operators SOURCENAMESPACE=openshift-marketplace'
Dec  2 13:16:28.522: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig create -f /tmp/e2e-test-olm-23440-fjdbm-config.json'
subscription.operators.coreos.com/test-operator created
Dec  2 13:16:41.000: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/Users/jianzhang/43-kubeconfig get -n e2e-test-olm-23440-fjdbm csv etcdoperator.v0.9.4 -o=jsonpath={.status.phase}'
Dec  2 13:16:51.002: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/Users/jianzhang/43-kubeconfig get -n e2e-test-olm-23440-fjdbm csv etcdoperator.v0.9.4 -o=jsonpath={.status.phase}'
STEP: Switch to common user to create the resources provided by the operator
Dec  2 13:16:52.604: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 process -f /var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/fixture-testdata-dir223261975/test/extended/testdata/olm/etcd-cluster.yaml -p NAME=example-etcd-cluster NAMESPACE=e2e-test-olm-23440-fjdbm'
Dec  2 13:16:53.912: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 create -f /tmp/e2e-test-olm-23440-fjdbm-config.json'
etcdcluster.etcd.database.coreos.com/example-etcd-cluster created
Dec  2 13:17:08.328: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get -n e2e-test-olm-23440-fjdbm etcdCluster example-etcd-cluster -o=jsonpath={.status}'
Dec  2 13:17:18.327: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get -n e2e-test-olm-23440-fjdbm etcdCluster example-etcd-cluster -o=jsonpath={.status}'
Dec  2 13:17:28.329: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get -n e2e-test-olm-23440-fjdbm etcdCluster example-etcd-cluster -o=jsonpath={.status}'
Dec  2 13:17:38.324: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get -n e2e-test-olm-23440-fjdbm etcdCluster example-etcd-cluster -o=jsonpath={.status}'
Dec  2 13:17:48.327: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get -n e2e-test-olm-23440-fjdbm etcdCluster example-etcd-cluster -o=jsonpath={.status}'
Dec  2 13:17:58.327: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get -n e2e-test-olm-23440-fjdbm etcdCluster example-etcd-cluster -o=jsonpath={.status}'
Dec  2 13:18:08.327: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get -n e2e-test-olm-23440-fjdbm etcdCluster example-etcd-cluster -o=jsonpath={.status}'
Dec  2 13:18:09.693: INFO: Running 'oc --namespace=e2e-test-olm-23440-fjdbm --config=/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile021121930 get pods -n e2e-test-olm-23440-fjdbm'
[AfterEach] [Feature:Platform] an end user use OLM
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/util/client.go:101
Dec  2 13:18:12.240: INFO: Deleted {user.openshift.io/v1, Resource=users  e2e-test-olm-23440-fjdbm-user}, err: <nil>
Dec  2 13:18:13.477: INFO: Deleted {oauth.openshift.io/v1, Resource=oauthclients  e2e-client-e2e-test-olm-23440-fjdbm}, err: <nil>
Dec  2 13:18:13.783: INFO: Deleted {oauth.openshift.io/v1, Resource=oauthaccesstokens  iLl7V_7MQbWI6I-mONM1GQAAAAAAAAAA}, err: <nil>
[AfterEach] [Feature:Platform] an end user use OLM
  /Users/jianzhang/goproject/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:152
Dec  2 13:18:13.783: INFO: Waiting up to 3m0s for all (but 100) nodes to be ready
STEP: Destroying namespace "e2e-test-olm-23440-fjdbm" for this suite.
Dec  2 13:18:15.117: INFO: Running AfterSuite actions on all nodes
Dec  2 13:18:15.117: INFO: Running AfterSuite actions on node 1

passed: (2m7s) 2019-12-02T05:18:15 "[Feature:Platform] an end user use OLM can subscribe to the etcd operator [Suite:openshift/conformance/parallel]"

Writing JUnit report to junit_e2e_20191202-051815.xml

1 pass, 0 skip (2m7s)
```

jianzhangbjz commented:
/retest

jianzhangbjz commented:
Squash and rebase it to the latest master branch, which that `Prometheus` issue fixed.

jianzhangbjz commented:
`/retest`

jianzhangbjz commented:
/lgtm

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
@bparees Do you know how to workaround this failure? Thanks very much! I have rebased on the latest master branch, which includes @wking's fix on this failure.
```console
[Feature:Prometheus][Conformance] Prometheus when installed on the cluster shouldn't report any alerts in firing state apart from Watchdog [Suite:openshift/conformance/parallel/minimal] expand_less | 1m34s
-- | --
fail [github.com/openshift/origin/test/extended/prometheus/prometheus_builds.go:161]: Expected     <map[string]error \| len:1>: {         "ALERTS{alertname!~\"Watchdog\|UsingDeprecatedAPIExtensionsV1Beta1\",alertstate=\"firing\"} >= 1": {             s: "promQL query: ALERTS{alertname!~\"Watchdog\|UsingDeprecatedAPIExtensionsV1Beta1\",alertstate=\"firing\"} >= 1 had reported incorrect results: model.Vector{(*model.Sample)(0xc0046793c0)}",         },     } to be empty


```

bparees commented:
/retest

bparees commented:
@jianzhangbjz not sure what fix you're referring to, I see an open BZ related to this being a flake:
https://bugzilla.redhat.com/show_bug.cgi?id=1779438

jianzhangbjz commented:
/test e2e-aws-fips

jianzhangbjz commented:
@bparees I mean this @wking PR: https://github.com/openshift/origin/pull/24248. Never mind, it seems like the corresponding test passed. @njhale @bparees Could you help give approval? Thanks!

bparees commented:
/unassign @bparees 
/assign @ecordell @shawn-hurley

jianzhangbjz commented:
/test e2e-aws-fips

jianzhangbjz commented:
@juzhao @brancz Do you know how to solve this failure?
```console
[Feature:Prometheus][Conformance] Prometheus when installed on the cluster shouldn't report any alerts in firing state apart from Watchdog [Suite:openshift/conformance/parallel/minimal] expand_less	1m30s
fail [github.com/openshift/origin/test/extended/prometheus/prometheus_builds.go:161]: Expected
    <map[string]error | len:1>: {
        "ALERTS{alertname!~\"Watchdog|UsingDeprecatedAPIExtensionsV1Beta1\",alertstate=\"firing\"} >= 1": {
            s: "promQL query: ALERTS{alertname!~\"Watchdog|UsingDeprecatedAPIExtensionsV1Beta1\",alertstate=\"firing\"} >= 1 had reported incorrect results: model.Vector{(*model.Sample)(0xc004cb2fa0)}",
        },
    }
to be empty
```

jianzhangbjz commented:
/test e2e-aws-fips

juzhao commented:
it seems like a flake

brancz commented:
You can use promecieus (not a typo, best to search on aos-devel how to use it) to query for the alert and see what’s wrong. Historically these have very rarely been false positives.

jianzhangbjz commented:
@shawn-hurley @ecordell @njhale Could you help add the `lgtm` label? Thanks!

shawn-hurley commented:
/approve

ecordell commented:
Can we do this test like our internal olm tests and create the catalog data as well? There is no guarantee that the community operator used here will always be available.

jianzhangbjz commented:
@ecordell Good point. I think I can modify it after this [PR 23186](https://github.com/openshift/origin/pull/23186)

jianzhangbjz commented:
@ecordell But, my key concern is that if we use the customized CatalogSource, they will be different test scenarios. And, the `etcd-operator` is a classic operator. So, I'd prefer to use this operator from the Community. Could you help add the `lgtm` label? Or any other concerns? Thanks!

emmajiafan commented:
/lgtm

jianzhangbjz commented:
/unsign @smarterclayton

jianzhangbjz commented:
/merge

jianzhangbjz commented:
Rebase it. @ecordell @njhale @emmajiafan   Could you help add the `lgtm` label? Thanks!

jianzhangbjz commented:
/test e2e-aws-fips

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

emmajiafan commented:
/lgtm
